### DISCUSSION TITLE
Key recovery

### DISCUSSION THREAD

**Discussion by Anushka3174** [decentralized-identity/veramo#438] [General]
Where is the key to the agent stored? I am trying to work on the credential recovery of the agent if the credentials are saved on the cloud.

**Comment by rado0x54** [comment]
This is a similar question to https://github.com/uport-project/veramo/discussions/425. As mentioned, Veramo is an pluggable agent framework and the (private) key storage method is dependent on the implementation of the KMS interface. The recovery mechanisms for the (respective) private key store / HSM would also be dependent on the plugin used.
